File: test_purs_ide.js
const net = require('net');
const client = new net.Socket();
client.connect(15894, '127.0.0.1', function () {
    console.log('Connected');
    client.write(JSON.stringify({
        command: "type",
        params: {
            file: "/home/namin/sources/purescript/src/Bill/Components/BillEditor.purs",
            search: "Bill"
        }
    }) + '\n');
});
client.on('data', function (data) {
    console.log('Received: ' + data);
    client.destroy();
});
client.on('close', function () {
    console.log('Connection closed');
});

File: test_purs_ide.py
import socket
import json

PORT = 15894
HOST = '127.0.0.1'

request = {
    "command": "type",
    "params": {
        "file": "/home/namin/sources/purescript/src/Bill/Components/BillEditor.purs",
        "search": "Bill"
    }
}

try:
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.connect((HOST, PORT))
        s.sendall(json.dumps(request).encode('utf-8') + b'\n')
        data = s.recv(4096)
    print(f"Received: {data.decode('utf-8')}")
except Exception as e:
    print(f"Error: {e}")
